pkg/deploy/tls.go

```diff
@@ -349,15 +350,22 @@ func K8sHandleCheTLSSecrets(deployContext *DeployContext) (reconcile.Result, err
 		}
 
 		domains := deployContext.CheCluster.Spec.K8s.IngressDomain + ",*." + deployContext.CheCluster.Spec.K8s.IngressDomain
-		if deployContext.CheCluster.Spec.Server.CheHost != "" && strings.Index(deployContext.CheCluster.Spec.Server.CheHost, deployContext.CheCluster.Spec.K8s.IngressDomain) == -1 && deployContext.CheCluster.Spec.Server.CheHostTLSSecret == "" {
+		if deployContext.CheCluster.Spec.Server.CheHost != "" && !strings.Contains(deployContext.CheCluster.Spec.Server.CheHost, deployContext.CheCluster.Spec.K8s.IngressDomain) && deployContext.CheCluster.Spec.Server.CheHostTLSSecret == "" {
 			domains += "," + deployContext.CheCluster.Spec.Server.CheHost
 		}
+
+		labels := ""
+		for labelName, labelValue := range GetLabels(deployContext.CheCluster, cheTLSSecretName) {
```

Comment:
I don't understand why we need to copy labels from one secret to another

Reply:
We do not copy anything. We get default set of labels for the TLS secret that job is going to create. We need the labels (actually only `instance` one, but to be consistent whole set is used) to make the secret cached in the Operator client.

Reply:
```suggestion
		for labelName, labelValue := range GetLabels(deployContext.CheCluster, DefaultCheFlavor(deployContext.CheCluster)) {
```